clinical_trials_api_scraper/utils/trial_model_utils.py

```diff
@@ -1,63 +1,101 @@
-    'resultsReportDate': 'ResultsFirstSubmitDate',
-    'contactEmail': 'PointOfContactEMail',
-    'clinicaltrialsUpdatedAt': 'DataVersion',
-    # '': 'StatusVerifiedDate',
-    # '': 'ResultsFirstSubmitQCDate',
-    # '': 'ResultsFirstPostDate',
-    # '': 'ResultsFirstPostDateType',
-    # '': 'CompletionDateType',
-    # '': 'WhyStopped',
+logger = logging.getLogger(__name__)
+
+# fields not in this map just get passed through
+RENAME_FIELDS_MAP = {
+    "NCTId": "id",
 }
-DATE_FIELDS = ['completionDate', 'clinicaltrialsUpdatedAt']
 
-INSTITUTION_FIELDS = ['orgName', 'orgType']
+ORGANIZATION_FIELDS = ["org_full_name", "org_class"]
 
 
 def extract_institution_from_trial(trial):
     institution = {}
-    for field in INSTITUTION_FIELDS:
+    for field in ORGANIZATION_FIELDS:
         institution[field] = trial.pop(field)
     return institution
 
 
-def split_institution_trial(full_trial):
+def split_organization_trial(full_trial):
     trial = full_trial.copy()
     institution = {}
-    for field in INSTITUTION_FIELDS:
+    for field in ORGANIZATION_FIELDS:
         value = trial.pop(field)
-        value = '' if value is None else value
+        value = "" if value is None else value
         institution[field] = value
     return institution, trial
 
 
 def trial_from_response_data(response_data):
-    trial_model = {
-        new_key: response_data.get(old_key) for new_key, old_key in RESPONSE_TO_TRIAL_MAP.items()
-    }
-    trial_model['isDelayed'] = trial_model.get('isDelayed') == 'Yes'
+    trial_model = {RENAME_FIELDS_MAP.get(k, k): v for k, v in response_data.items()}
+
+    for k, v in trial_model.items():
+        if not isinstance(v, str):
+            continue
+
+        # convert date columns
+        if "Date" in k and v is not None:
+            trial_model[k] = parser.parse(v)
+            continue
+
+        # convert anything that looks boolean into a bool
+        if v == "Yes":
+            trial_model[k] = True
+            continue
+        elif v == "No":
+            trial_model[k] = False
+            continue
+
+        # convert integers (I'm not aware of any floats in the schema)
+        try:
+            trial_model[k] = int(v)
+            continue
+        except ValueError:
+            pass
+
+    return dict_to_snake_case(trial_model)
+
 
-    for time_field in DATE_FIELDS:
-        if trial_model.get(time_field):
-            trial_model[time_field] = parser.parse(
-                trial_model[time_field]).isoformat()
-    return trial_model
+def add_computed_fields(trial):
+    """Adds Computed fields to a trial
+
+    Fields related to https://clinicaltrials.gov/ct2/manage-recs/fdaaa
+    should_have_results: the trial should have reported results by now
+    is_late: the trial was late reporting results, but is not missing
+    is_missing: the trial results should have been reported by now but are missing
+    """
+    one_year = dt.timedelta(365)
+    one_year_ago = dt.datetime.now() - one_year
```

Comment:
Don't think it matters but if we want to be more precise we could pull in the DataVersion date rather than now.

Reply:
Good call! Fixed.